### PR TITLE
Added ICD-10 data with tests and documentation

### LINKED ISSUE #3: Add ICD-10 data
Add the ICD-10 data from 2011/12 to 2023/24 available at https://digital.nhs.uk/data-and-information/publications/statistical/hospital-admitted-patient-care-activity. Follow the implementation of the SNOMED data in #2.

- [x] Tidy the data in `data-raw/icd10_code_usage.R`
- [x] Add the data to the R package using at the end of the `data-raw/icd10_code_usage.R` script
    
    ```R
    usethis::use_data(
      icd10_usage,
      compress = "bzip2",
      overwrite = TRUE
    )
    ```

- [x] Document the data in `R/data.R` using https://roxygen2.r-lib.org/, for an example see https://github.com/ebmdatalab/codeusage/blob/d2c42e567f3f34d447fa7089e677373e9e53fbee/R/data.R#L4-L22
- [x] Run `devtools::document(roclets = c('rd', 'collate', 'namespace', 'vignette'))` or the keyboard shortcut for Updating the documentation
- [x] Write some tests to check the data (e.g., date range, sum of usage) to `tests/testthat/test-data.R`, for examples see the existing tests https://github.com/ebmdatalab/codeusage/blob/d2c42e567f3f34d447fa7089e677373e9e53fbee/tests/testthat/test-data.R#L30-L52
- [x] Check that all tests are passing using `devtools::test()` or the shortcut for testing the package
- [x] Add a section describing the data and how to use it to `README.Rmd` and run the RMarkdown document to update `README.md`
- [x] Run `devtools::check()` to check the package, this will take around 1 minute and there should only be one note like this:

    ```
    ── R CMD check results ─────────────────────────────── codeusage 0.0.1 ────
    Duration: 47s
    
    ❯ checking installed package size ... NOTE
        installed size is 19.7Mb
        sub-directories of 1Mb or more:
          data  19.7Mb
    
    0 errors ✔ | 0 warnings ✔ | 1 note ✖
    R CMD check succeeded
    ```

### PR DESCRIPTION
**Pull request by atamborska**
Closes #3

### CODE REVIEW

File: R/data.R [hunk 1]
```diff
@@ -20,3 +20,21 @@ NULL
...
 #' @source <https://digital.nhs.uk/data-and-information/publications/statistical/mi-snomed-code-usage-in-primary-care>
 "snomed_usage"
+
+#' ICD-10 Code Usage from Hospital Admitted Patient Care Activity in England
+#'
+#' Yearly summary of ICD-10 code usage from 1st April 2013 to 31st March 2024.
+#' Restricted codes. for which annual usage is not published, have been removed from the dataset.
+#'
+#' @format ## `icd10_usage`
+#' A data frame with 135,951 rows and 6 columns:
```
Comment (milanwiedemann) [inline R/data.R:30]:
```suggestion
#' @format A data frame with 135,951 rows and 6 columns:
```

File: R/data.R [hunk 2]
```diff
@@ -20,3 +20,21 @@ NULL
...
+#' \describe{
+#'   \item{start_date}{Start date of code usage count}
+#'   \item{end_date}{End date of code usage count}
+#'   \item{icd10_code}{ICD-10 COde.
```
Comment (milanwiedemann) [inline R/data.R:34]:
```suggestion
#'   \item{icd10_code}{ICD-10 Code.
```

File: data-raw/icd10_code_usage.R [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+icd10_usage <- icd10_code_usage_urls |>
+  map(get_icd10_data) |>
+  bind_rows(.id = "nhs_fy") |>
+  separate(nhs_fy, c("start_date", "end_date"), "to") |>
+  mutate(
+    start_date = as.Date(
+      paste0("20", str_extract_all(start_date, "\\d+"), "-04-01")),
+    end_date = as.Date(
+      paste0("20", str_extract_all(end_date, "\\d+"), "-03-31")),
```
Comment (milanwiedemann) [inline data-raw/icd10_code_usage.R:128]: Did you find out whether these are the correct start and end dates for this dataset?

File: README.html
Comment (milanwiedemann) [inline README.html:1]: We dont need this file, only the `README.md` file is important to keep, as is the `README.Rmd` that is used to generate the file

**Review by milanwiedemann — APPROVED** [review]
WONDERFUL! I added very minor suggestions, main thing is that we should remove the `README.html` file as it isn't needed

File: data-raw/icd10_code_usage.R (hunk 3 above)
Comment (atamborska) [reply data-raw/icd10_code_usage.R:128]: The admitted patient activity represents funancial year, but to be perfectly accurate, the financial year would run 6 April 2023 to on 5 April 2024

**Pull request merged**